From 1493bbfab9b624822b3f06c274c6e8d53e7acf8e Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 17 Nov 2017 16:15:45 +0800
Subject: [PATCH 2829/5242] MLK-16887-2: ARM64: dts: enable spdif2 and add
 sound card for hdmi arc

commit  c32b94dbb5b17fdfbb937584f976ddd686850d64 from
https://source.codeaurora.org/external/imx/linux-imx.git

enable spdif2 and add sound card for hdmi arc for mscale

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |   17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 7deb8a9..fcc0bf1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -82,7 +82,7 @@
 		compatible = "fsl,imx-audio-cdnhdmi";
 		model = "imx-audio-hdmi";
 		audio-cpu = <&sai4>;
-		protocol = <0>;
+		protocol = <1>;
 	};
 
 	sound-spdif {
@@ -92,6 +92,13 @@
 		spdif-out;
 		spdif-in;
 	};
+
+	sound-hdmi-arc {
+		compatible = "fsl,imx-audio-spdif";
+		model = "imx-hdmi-arc";
+		spdif-controller = <&spdif2>;
+		spdif-in;
+	};
 };
 
 &clk {
@@ -613,6 +620,14 @@
 	status = "okay";
 };
 
+&spdif2 {
+	assigned-clocks = <&clk IMX8MQ_CLK_SPDIF2_SRC>,
+			<&clk IMX8MQ_CLK_SPDIF2_DIV>;
+	assigned-clock-parents = <&clk IMX8MQ_AUDIO_PLL1_OUT>;
+	assigned-clock-rates = <0>, <24576000>;
+	status = "okay";
+};
+
 &gpu_pd {
 	power-supply = <&sw1a_reg>;
 };
-- 
1.7.9.5

